crates/uv/src/commands/python/install.rs

```diff
@@ -302,114 +312,136 @@ pub(crate) async fn install(
             .expect("We should have a bin directory with preview enabled")
             .as_path();
 
-        let target = bin.join(installation.key().versioned_executable_name());
+        let targets = if (default || is_default_install)
+            && first_request.matches_installation(installation)
+        {
+            vec![
+                installation.key().executable_name_minor(),
+                installation.key().executable_name_major(),
+                installation.key().executable_name(),
+            ]
+        } else {
+            vec![installation.key().executable_name_minor()]
+        };
 
-        match installation.create_bin_link(&target) {
-            Ok(()) => {
-                debug!(
-                    "Installed executable at {} for {}",
-                    target.simplified_display(),
-                    installation.key(),
-                );
-                changelog.installed.insert(installation.key().clone());
-                changelog
-                    .installed_executables
-                    .entry(installation.key().clone())
-                    .or_default()
-                    .push(target.clone());
-            }
-            Err(uv_python::managed::Error::LinkExecutable { from: _, to, err })
-                if err.kind() == ErrorKind::AlreadyExists =>
-            {
-                debug!(
-                    "Inspecting existing executable at {}",
-                    target.simplified_display()
-                );
+        for target in targets {
+            let target = bin.join(target);
```

Comment:
Is everything from here down mostly just an indent of the existing logic?

Reply:
Yeah, and some tweaks from rebases though I've reduced those a couple times.

> This pull request is best viewed with [whitespace hidden](https://github.com/astral-sh/uv/pull/8650/files?diff=unified&w=1)